priv/templates/phx.gen.html/index.html.eex

```diff
@@ -13,14 +12,13 @@
     <tr>
 <%= for {k, _} <- schema.attrs do %>      <td><%%= <%= schema.singular %>.<%= k %> %></td>
 <% end %>
-      <td class="text-right">
-        <span><%%= link "Show", to: Routes.<%= schema.route_helper %>_path(@conn, :show, <%= schema.singular %>), class: "btn btn-default btn-xs" %></span>
-        <span><%%= link "Edit", to: Routes.<%= schema.route_helper %>_path(@conn, :edit, <%= schema.singular %>), class: "btn btn-default btn-xs" %></span>
-        <span><%%= link "Delete", to: Routes.<%= schema.route_helper %>_path(@conn, :delete, <%= schema.singular %>), method: :delete, data: [confirm: "Are you sure?"], class: "btn btn-danger btn-xs" %></span>
+      <td>
+        <%%= link "Show", to: Routes.<%= schema.route_helper %>_path(@conn, :show, <%= schema.singular %>) %>
```

Comment:
Regarding this SO answer it isn't valid HTML https://stackoverflow.com/questions/6393827/can-i-nest-a-button-element-inside-an-a-using-html5

Could't be `button` used instead of `link` ?

```
<%%= button "Show", to: ... %>
```

Reply:
Hm I have been using `<button>` inside `<a>` for ages (the other way around you may have UX problems).

The problem with the button tag is that it doens't feature a href attribute and it can only be made to work like links with javascript onclick which is really bad practise or with a http get form wrapper (which would require something like `link_button to:` helper

Reply:
hmm.. you are totally right, `button to: ..` will not work.. the cleanest way would be to use a class actually :D

Reply:
@josevalim what do you think of adding `phx-gen_link-btn`, `phx-gen_textfield` classes to the auto generated markup, so people can add they styles how they want to. In case they remove that markup, it does not matter anyway..

Reply:
You can easily target the fields with regular css selectors.
However having a button link may still be desireable and the solution should be a small helper that generates a HTTP GET form

Reply:
If we are going to be filling it with our own classes, then I would rather stick with bootstrap. The goal I see with using Marx is to have fewer classes and reduce the size of the default CSS, so we should stick with those goals in mind.

Reply:
My intention was not to bind the markup to a specific css framework, instead by adding `.phx-` classes it would just declare the semantics, which are there just to describe the generated context.

Just a simple example of what I mean with *semantics*

```
<form class="phx-edit-form phx-model-user">
  <div class="phx-textfield phx-field-username">
    <label>Username</label>
    <input />
  <div>
</form>
```

My intention was not to provide production ready styles and rebuild bootstrap at all. Instead, I can imagine that a lot of people would like to provide a simple css files to theme phoenix generated markup.

Reply:
However, this idea just came to my mind, but I actually don't think that this belongs here... sorry for the noise :)